rust/kernel/tracepoint.rs

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: GPL-2.0
+
+// Copyright (C) 2024 Google LLC.
+
+//! Logic for tracepoints.
+
+/// Declare the Rust entry point for a tracepoint.
+#[macro_export]
+macro_rules! declare_trace {
+    ($(#[$attr:meta])* $pub:vis fn $name:ident($($argname:ident : $argtyp:ty),* $(,)?);) => {
+        $crate::macros::paste! {
+            $( #[$attr] )*
+            #[inline(always)]
+            $pub fn [< trace_ $name >]($($argname : $argtyp),*) {
```

Comment:
Not a big deal. But ideally, this could be a `#[nomangle]` and `extern "C"` function so that C code can use tracepoints declared by Rust.

Reply:
I don't think there's any benefit from this. The Rust support already requires that you _also_ have a `TRACE_EVENT` invocation on the C side, so C code should just call into the equivalent static inline C method generated by that macro.

Reply:
Oh, I see, this is only declaring the tracepoint, you still need C side to define the tracepoint.